### ISSUE TITLE
Weird behavior of X (close window) button

### ISSUE THREAD

**Issue by NSurtsev** [microsoft/terminal#12933]
### Windows Terminal version

1.12.10732.0

### Windows build number

19044.1645

### Steps to reproduce

Enable "always display icon in system tray" and then close the window of Terminal

### Expected Behavior

Window is closed, but Terminal is still running with icon in system tray (the same behavior for any program in system tray, that's why programs usually have option in Settings to be set to system tray!)
To close Terminal completely you should go to system tray and do it manually with right-click on icon and "close" (or smth)

### Actual Behavior

Nevertheless I have Terminal in system tray, after I close the window... the Terminal is simply closed...

**Comment by zadjii-msft** [comment]
If you hover over the icon in the tray, does it disappear/? (I think I've seen this in the past on Debug builds. We may want to fix that if it's present in release builds as well)

When you right click on the tray icon, and select "windows >", are there any listed there?

**Comment by NSurtsev** [comment]
> If you hover over the icon in the tray, does it disappear/? (I think I've seen this in the past on Debug builds. We may want to fix that if it's present in release builds as well)
> 
> When you right click on the tray icon, and select "windows >", are there any listed there?

It doesn't disappear.
"Windows" shows my current Terminal windows (usually 1 or 2 if I lauch Quake mode too).
I considered it just wrong implementation of functions. There is also option in Settings "hide Terminal in system tray when it is minimized". I don't know why anyone could need this option enabled with "Minimize" button (we usually minimize to still see program on taskbar and be able to just Alt-TAB, if needed), but that's actually how I want it to work (and how it should work!) with "Close" button and system tray enabled by default!

**Comment by NSurtsev** [comment]
> If you hover over the icon in the tray, does it disappear/? (I think I've seen this in the past on Debug builds. We may want to fix that if it's present in release builds as well)
> 
> When you right click on the tray icon, and select "windows >", are there any listed there?

It doesn't disappear.
"Windows" shows my current Terminal windows (usually 1 or 2 if I lauch Quake mode too).
I considered it just wrong implementation of function. There is option in Settings "hide Terminal in system tray when it is minimized". I don't know why anyone could need this option enabled with "Minimize" button (we usually minimize to still see program on taskbar and be able to Alt-TAB), but that's actually how it should work with "Close" button and system tray enabled!

**Comment by zadjii-msft** [comment]
Ah yea, sorry, I misread that. You want close to minimize/hide to the tray, rather than close the window. I'm pretty sure we've had a discussion about this in the paste, lemme find it...

**Comment by NSurtsev** [comment]
> Ah yea, sorry, I misread that. You want close to minimize/hide to the tray, rather than close the window. I'm pretty sure we've had a discussion about this in the paste, lemme find it...

Exactly! This is default when we are talking about programs, showing on system tray! It's very strange that Terminal doesn't behave this way. I checked for issues for last half a year, but didn't see discussion about it, maybe I missed it.

**Comment by JVimes** [comment]
This is worded roughly as "Window close button minimizes instead of terminating" in KeePass. Maybe that's useful for us.

**Comment by only-su** [comment]
Coming from linux the ability to have the terminal always accessible (i.e. it never really closes) is a most wealcome feature
